tests/test_georaster.py

```diff
@@ -584,6 +634,27 @@ def test_reproject(self, example: str) -> None:
 
             plt.show()
 
+        # - Check that if mask is modified afterwards, it is taken into account during reproject - #
+        # Create a raster with (additional) random gaps
+        r_gaps = r.copy()
+        data_1d = r_gaps.data.reshape(-1)
+        valids = np.where(~np.ma.atleast_1d(data_1d).mask)[0]
+        assert len(valids) > 0  # sanity check
+        nsamples = 200
+        data_1d[np.random.choice(valids, nsamples, replace=False)] = np.ma.masked
+        r_gaps.data = data_1d.reshape(r.data.shape)
+        assert np.sum(r_gaps.data.mask) - np.sum(r.data.mask) == nsamples  # sanity check
+
+        # reproject raster, and reproject mask. Check that both have same number of masked pixels
+        # TODO: should test other resampling algo
+        r_gaps_reproj = r_gaps.reproject(dst_res=dst_res, resampling="nearest")
+        mask = gu.Raster.from_array(
+            r_gaps.data.mask.astype("uint8"), crs=r_gaps.crs, transform=r_gaps.transform, nodata=None
+        )
+        mask_reproj = mask.reproject(dst_res=dst_res, dst_nodata=255, resampling="nearest")
+        tot_masked_true = np.sum(mask_reproj.data.mask) + np.sum(mask_reproj.data == 1)
```

Comment:
Can you comment a bit more why you sum those two? Also on the creating of random gaps above, some functions such as `atleast_1d` are not so common, so it'd avoid one of us Googling that in the future if we need to re-work/understand these lines :slightly_smiling_face:.

Side note: I also typically use `np.count_nonzero` and not `np.sum`. Here it's unimportant, but I thought it was always a detail until I realized that `np.count_nonzero` always outputs an integer, which is quite practical to make `typing` happy and consistent in some instances (who knew I'd eventually become typing-enthusiast :laughing: )